controllers/components/kserve/kserve_controller.go

```diff
@@ -19,40 +19,85 @@ package kserve
 import (
 	"context"
 
-	"k8s.io/apimachinery/pkg/runtime"
+	templatev1 "github.com/openshift/api/template/v1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 
 	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	featuresv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/features/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/render"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/updatestatus"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
 )
 
-// KserveReconciler reconciles a Kserve object.
-type KserveReconciler struct {
-	client.Client
-	Scheme *runtime.Scheme
-}
+// NewComponentReconciler creates a ComponentReconciler for the Dashboard API.
+func NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ComponentReconcilerFor(mgr, componentsv1.KserveInstanceName, &componentsv1.Kserve{}).
+		// operands - owned
+		Owns(&corev1.Secret{}).
+		Owns(&corev1.Service{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&rbacv1.ClusterRole{}).
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		Owns(&templatev1.Template{}).
+		Owns(&featuresv1.FeatureTracker{}).
```

Comment:
do not think we should own FTer, should we?

Reply:
since FT are generated by the component, then it should own them

Reply:
The FTs that are currently created when using kserve appear to be owned by the DSC, so I presumed they should be owned by the Kserve CR now instead. TBH I don't really understand FTs though, so I might be missing many details.

Reply:
After trying to understand FTs, and how they're used for KServe, I _think_ they should be owned by the Kserve CR, but _maybe_ there should be a finalizer set so that `removeServerlessFeatures` or `removeServiceMeshConfigurations` are called so that any `cleanups` defined in the Features will be processed if the Kserve instance is deleted? At the same time though, I don't believe there are any cleanups defined for any of these, so maybe that's not useful.

@zdtsw I'm interested to hear your thoughts on how these should be handled, and what we should do about ownership for the FTs.

Reply:
My understanding is that:
- `removeServerlessFeatures` and `removeServiceMeshConfigurations` are functions meant to reacts on changes to the KServer component configuration i.e. if you make the knative serving setup `Managed` or `Removed`, some resources should be removed/amended.
- when the Kserver CR gets deleted, then I'd expect that the related FTs are also deleted, causing all the resources associated to be deleted.
- If the above is correct, we would not need to add a finalizer (preferred option) but that should be confirmed :) 

Ideally as second step (and on another PR), I'd like to remove those FT and just go to the component pipeline so we can benefit from the caching and the other improvement introduced as part of the component pipeline.

Reply:
@lburgazzoli I have the same understanding as you I think; my thought about a finalizer was really just because _technically_ a FT shouldn't be left to be garbage collected by Kubernetes: there's the possibility of "cleanup" actions defined as part of the feature, that technically should be processed as part of the deletion, before the FT should be removed. I don't believe there are any cleanup actions for any of these Features that Kserve creates, so I don't think there's any need to use the Feature machinery to remove the FT when the Kserve is removed: it should be fine in my understanding to let K8s gc it.

I think I agree with your "second step" proposal too: I think that would be something to try in a future PR which should hopefully simplify things here.

Reply:
back and forth in this PR several times and never had time to finalize my words, so i tried it in a short format this time and hope it wont get lost again.....
we should have the ownership set on these FTers (which we did not do that in the incubation yet) thats why i was not sure if we wanna do it or with a follow-up PR.  it might need some cleanups, say we add ownership, the needs for .spec.source should not exist any longer. but no rush to get all done for now.
in another short-short:  i am fine with adding `Owns(&featuresv1.FeatureTracker{}).`

Reply:
Ack, thanks for the info @zdtsw! :+1: